packages/nextjs/src/tools/codegen/import-map.ts

```diff
@@ -0,0 +1,421 @@
+        error
+      );
+      throw error;
+    }
+  };
+};
+
+/**
+ * Builds file contents for component map based on the default template
+ * @param {Map<string, ImportModule>} indexedImportMap map to be processed into final component-map.ts file
+ * @returns {string} file code for component-map.ts
+ */
+export const nextJsMapTemplate = (indexedImportMap: Map<string, ModuleExports>) => {
+  const getSingleImport = (originalName: string, aliasName: string) => {
+    return originalName !== aliasName ? `${originalName} as ${aliasName}` : originalName;
+  };
+
+  const convertNamedImports = (namedImports: ExportDefinition[]) => {
+    return namedImports
+      .map((entry) => {
+        return getSingleImport(entry.name, entry.value);
+      })
+      .join(', ');
+  };
+
+  const outputExportEntries = (exports: ExportDefinition[]) => {
+    return exports.length
+      ? exports
+          .map(
+            (namedExport) => `      { name: '${namedExport.name}', value: ${namedExport.value} }`
+          )
+          .join(',\n')
+      : '';
+  };
+
+  const importStatements: string[] = [];
+  const importMapArray = Array.from(indexedImportMap);
+
+  // get import map entries after
+  const finalImportMap: ImportMapEntry[] = importMapArray.map(([modulePath, imports]) => {
+    const defaultExports = Array.from(imports.defaultExports).map(([name, value]) => {
+      return {
+        name,
+        value,
+      };
+    });
+    const namedExports = Array.from(imports.namedExports).map(([name, value]) => {
+      return {
+        name,
+        value,
+      };
+    });
+    return {
+      module: modulePath,
+      defaultExports,
+      namedExports,
+    };
+  });
+
+  finalImportMap.forEach((entry) => {
+    if (entry.namedExports.length > 0) {
+      importStatements.push(
+        `import { ${convertNamedImports(entry.namedExports)} } from '${entry.module}';`
+      );
+    }
+    if (entry.defaultExports.length > 0) {
+      Array.from(entry.defaultExports).forEach((defaultExportEntry) => {
+        importStatements.push(
+          `import ${getSingleImport(defaultExportEntry.name, defaultExportEntry.value)} from '${
+            entry.module
+          }';`
+        );
+      });
+    }
+  });
+
+  return `// This file is auto-generated by the Sitecore Content SDK.
+// Below are built-in Content SDK imports neccessary for the import map
+import { combineImportEntries, defaultImportEntries } from '@sitecore-content-sdk/nextjs/tools';
+// end of built-in imports
+  
+${importStatements.join('\n')}
+    
+const importMap = [
+${finalImportMap
+  .map((entry) =>
+    [
+      '  {',
+      `    module: '${entry.module}',`,
+      '    exports: [',
+      outputExportEntries(entry.namedExports),
+      outputExportEntries(entry.defaultExports),
+      '    ]',
+      '  }',
+    ].join('\n')
+  )
+  .join(',\n')}
+];
+
+export default combineImportEntries(defaultImportEntries, importMap);
+`;
+};
+
+export const combineImportEntries = (
```

Comment:
Here you have just two arguments, but this function might be useful when we have multiple import maps, e.g, customer created separate map for each subproject.
Let's change incoming arguments to ...args: Array<ImportEntry[]> and allow to combine many maps